src/main/webapp/resources/js/pages/projects/samples/components/RemoveModal.jsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import { Col, Divider, List, Modal, Row, Typography } from "antd";
+import { useRemoveMutation } from "../../../../apis/projects/samples";
+import LockedSamplesList from "./LockedSamplesList";
+import AssociatedSamplesList from "./AssociatedSamplesList";
+
+/**
+ * React Element to display a modal with sample to be removed from the current
+ * project.
+ *  - Will display samples that are locked - cannot be removed.
+ *  - Will display associated samples that cannot be removed from this project.
+ * @param {array} samples - list of samples to remove from the current project
+ * @param {boolean} visible - whether the modal is currently visible on the page
+ * @param {function} onComplete - action to perform after the remove is complete
+ * @param {function} onCancel - action to perform if the remove is cancelled.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function RemoveModal({
+  samples,
+  visible,
+  onComplete,
+  onCancel,
+}) {
+  const [removeSamples, { isLoading }] = useRemoveMutation();
+
+  const onOk = async () => {
+    try {
+      const response = await removeSamples(
+        samples.valid.map((sample) => sample.id)
+      );
+      onComplete();
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  return (
+    <Modal
+      title={i18n("RemoveModal.title")}
+      className="t-remove-modal"
+      visible={visible}
+      onCancel={onCancel}
+      onOk={onOk}
+      okText={i18n("RemoveModal.okText")}
+      okButtonProps={{
+        loading: isLoading,
+      }}
+      width={600}
+    >
+      <Row gutter={[16, 16]}>
+        <Col span={24}>
+          <List
+            size="small"
+            bordered
+            header={<Typography.Text>Samples to be removed</Typography.Text>}
```

Comment:
Internationalize

Reply:
Updated in 0ca6f86ce463ac3d058e5a0f04aba2236bf797fb

Reply:
Sorry to be a pain but can you change the translation key (`RemoveModal.good`) so it's clear what it is?